File: 128. Longest Consecutive Sequence/128.py
class Solution:
    def longestConsecutive(self, nums: list[int]) -> int:
        hash = set(nums)
        long = 0

        for i in hash:
            if (i-1) not in hash:
                aux = 1
                while (i+aux) in hash:
                    aux += 1
                long = max(long, aux)
        return long


S = Solution()
print(S.longestConsecutive([1,7,10]))

File: 128. Longest Consecutive Sequence/128.cs
public class Solution {
    public int LongestConsecutive(int[] nums) {
        HashSet<int> hash = new HashSet<int>(nums);
        int longest = 0;
        foreach (int n in hash){
            if (!hash.Contains(n-1)){
                int count = 1;
                while (hash.Contains(n+count)){
                    count++;
                }
                longest = Math.Max(longest, count);
            }
        }
        return longest;
    }
}